### PR TITLE
Adding redux

### PR DESCRIPTION
**Pull request by ZacMiddleton**
adding redux to application. entire app not converted to redux yet in process.

### CODE REVIEW

File: src/components/ConversionCalculator/ConversionCalculator.styles.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import styled from "styled-components";
+
+export const InputWrapper = styled.div`
+  // width: 270px;
```
Comment (yoloOnTheBattlefield) [inline src/components/ConversionCalculator/ConversionCalculator.styles.jsx:4]: remove this

File: src/components/CryptoTable/index.jsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            const colorTwo = colorGeneratorTwo[Math.floor(Math.random() * 4)];
+            return (
+              <ListItem key={item.id}>
+                <CoinTitle to={`/coin/${item.id}`}>
+                  <img src={item.image} alt="" />{" "}
+                  {`${item.name} (${item.symbol.toUpperCase()})`}
+                </CoinTitle>
+                <p>{`${currencySymbol(
+                  currency
+                )}${new Intl.NumberFormat().format(item.current_price)}`}</p>
```
Comment (yoloOnTheBattlefield) [inline src/components/CryptoTable/index.jsx:157]: put this in a utility fn

File: src/pages/Coins/Coins.jsx [hunk 3]
```diff
@@ -1,8 +1,102 @@
...
+          />
+        )}
+      </Container>
+    );
+  }
 }
 
-export default Coins;
\ No newline at end of file
+const mapStateToProps = (state) => ({
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins/Coins.jsx:98]: use the redux hooks

File: src/store/coinData/index.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { GET_COINDATA_SUCCESS } from './actions';
+
+const initialState = {};
+
+const coinDataReducer = (state = initialState, action) => {
+    console.log(state);
```
Comment (yoloOnTheBattlefield) [inline src/store/coinData/index.jsx:6]: remove this

File: src/App.jsx [hunk 5]
```diff
@@ -1,69 +1,68 @@
...
-    </>
-  );
-};
+  // const getCoinData = (info) => {
+  //   setCoinInfo(info.data)
+  // };
 
-function App() {
-  const [theme, setTheme] = useState("dark");
+  console.log(coinData);
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:27]: remove this

File: src/store/coinData/actions.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import axios from "axios";
+
+export const GET_COINDATA_SUCCESS = "GET_COINDATA_SUCCESS"
+export const GET_COINDATA_ERROR = "GET_COINDATA_ERROR"
+
+export const getCoinData = (currency, page, perPage) => async dispatch => {
+    console.log("called");
```
Comment (yoloOnTheBattlefield) [inline src/store/coinData/actions.jsx:7]: remove this

File: src/App.jsx [hunk 7]
```diff
@@ -1,69 +1,68 @@
...
-export default App;
+const mapStateToProps = (state) => ({
+  theme: state.theme,
+  currency: state.currency,
+  coinData: state.coinData.data,
+})
+
+const mapDispatchToProps = {
+  getCoinData,
+}
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:66]: use the hooks

File: src/components/BtcBarChart/index.jsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        <p>{getFormattedDate()}</p>
+        {barData && <Bar data={data} options={options} />}
+      </Container>
+    </ChartWrapper>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  currency: state.currency,
+})
```
Comment (yoloOnTheBattlefield) [inline src/components/BtcBarChart/index.jsx:105]: use the hooks

File: src/components/BtcLineChart/index.jsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+        <p>{getFormattedDate()}</p>
+        {lineData && <Line ref={chartRef} options={options} data={data} />}
+      </Container>
+    </ChartWrapper>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  currency: state.currency,
+})
```
Comment (yoloOnTheBattlefield) [inline src/components/BtcLineChart/index.jsx:116]: use the hooks

File: src/components/CryptoTable/index.jsx [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+          })}
+        </InfiniteScroll>
+      </StyledList>
+    </TableContainer>
+  );
+}
+
+const mapStateToProps = (state) => ({
+  currency: state.currency,
+})
```
Comment (yoloOnTheBattlefield) [inline src/components/CryptoTable/index.jsx:219]: use the hooks

File: src/components/Root/index.jsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        <Outlet />
+      </div>
+    </>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  currency: state.currency,
+  theme: state.theme,
+});
```
Comment (yoloOnTheBattlefield) [inline src/components/Root/index.jsx:77]: use the hooks

File: src/pages/CoinPage/CoinPage.jsx [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+        )}
+      </Container>
+      {coinInfo && <CoinPageChart coinId={coinId} currency={cur} timeline={timeline} />}
+    </>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  currency: state.currency,
+})
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage/CoinPage.jsx:316]: use the hooks

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/pages/Coins/Coins.jsx (hunk 3 above)
Comment (ZacMiddleton) [reply src/pages/Coins/Coins.jsx:98]: coins is a class component, im going to convert it to hooks and make a separate pull request for that

**Pull request merged**